pkg/cmd/secret/set/set.go

```diff
@@ -153,6 +157,7 @@ func NewCmdSet(f *cmdutil.Factory, runF func(*SetOptions) error) *cobra.Command
 	cmd.Flags().StringVarP(&opts.Body, "body", "b", "", "The value for the secret (reads from standard input if not specified)")
 	cmd.Flags().BoolVar(&opts.DoNotStore, "no-store", false, "Print the encrypted, base64-encoded value instead of storing it on Github")
 	cmd.Flags().StringVarP(&opts.EnvFile, "env-file", "f", "", "Load secret names and values from a dotenv-formatted `file`")
+	cmd.Flags().StringVarP(&opts.Application, "app", "a", "", "Set the application for a secret")
```

Comment:
We introduced a flag helper called [`StringEnumFlag`](https://github.com/cli/cli/blob/6f66a1d265d1b4b1e3d2e742decfd474932bc168/pkg/cmdutil/flags.go#L27) which I think would be helpful to use here. I think we will need to pull in the latest trunk to have access to it though. We can use it to restrict input to only valid application values.

Reply:
If it isn't too much trouble could we also apply that flag helper to the `Visibility` flag as well?

Reply:
Added in https://github.com/cli/cli/pull/5134/commits/47182ab0098e82e8dbd635f20d66102c40868114



> If it isn't too much trouble could we also apply that flag helper to the `Visibility` flag as well?
Looks like that was already added in #5221